intelmq/lib/upgrades.py

```diff
@@ -662,6 +663,32 @@ def v300_pipeline_file_removal(defaults, runtime, harmonization, dry_run):
     return messages if messages else changed, defaults, runtime, harmonization
 
 
+def v301_deprecations(defaults, runtime, harmonization, dry_run):
+    """
+    Deprecate malwaredomains parser and collector
+    """
+    found_malwaredomainsparser = []
+    found_malwaredomainscollector = []
+    changed = None
+    messages = []
+    for bot_id, bot in runtime.items():
+        if bot["module"] == "intelmq.bots.parsers.malwaredomains.parser":
+            found_malwaredomainsparser.append(bot_id)
+        if bot["module"] == "intelmq.bots.collectors.http.collector":
+            if "http_url" not in bot["parameters"]:
+                continue
+            if 'malwaredomains.com' in bot["parameters"]["http_url"]:
```

Comment:
```suggestion
            if bot["parameters"]["http_url"].startswith('http://malwaredomains.com'):
```
CodeQL detects here a *Incomplete URL substring sanitization*, which is technically right, in practice - probably - not relevant, but just to be sure.

Reply:
The URL listed in the documentation is `http://mirror1.malwaredomains.com/files/domains.txt` - I thought there might be other similar mirrors in use somewhere, but maybe its enought to just check for the URL as listed in the documentation.

Reply:
Just checking for the documented URL is OK I think